### PR TITLE
handle super- and subscript text in PdfMapper and FastPdfMapper

### PR DESCRIPTION
**Pull request by goekay**
i hope i used the correct classes and conventions while implementing it, since i am not that familiar with the codebase.

another remark: i used the 1/3 of the font size to raise or lower the text (https://github.com/opensagres/xdocreport/pull/276/commits/cec56b863b3aee53c3b7f208d1b974489f5cf21b#diff-ba6c8c935dbf8bd96b619b1f7a6a177eR663 and https://github.com/opensagres/xdocreport/pull/276/commits/cec56b863b3aee53c3b7f208d1b974489f5cf21b#diff-ba6c8c935dbf8bd96b619b1f7a6a177eR666) after experimenting with some other values. i am not that sure that using a fixed ratio of the font size is a sound approach.

### CODE REVIEW

**Comment by angelozerr** [comment]
Please add a sample which causes problem.

**Comment by goekay** [comment]
should i just upload a docx file which uses super- and subscripts or do you mean java test cases?

**Comment by angelozerr** [comment]
> should i just upload a docx file which uses super- and subscripts or do you mean java test cases?

Please add your (simple docx) in 

 * https://github.com/opensagres/xdocreport/blob/master/thirdparties-extension/fr.opensagres.poi.xwpf.converter.pdf/src/test/java/org/apache/poi/xwpf/converter/core/AbstractXWPFPOIConverterTest.java
 * https://github.com/opensagres/xdocreport/blob/master/thirdparties-extension/fr.opensagres.poi.xwpf.converter.pdf.itext5/src/test/java/org/apache/poi/xwpf/converter/core/AbstractXWPFPOIConverterTest.java

Thanks!

**Comment by goekay** [comment]
i would gladly do it as you described, but the [test cases](https://github.com/opensagres/xdocreport/tree/master/thirdparties-extension/fr.opensagres.poi.xwpf.converter.pdf.itext5/src/test/java/org/apache/poi/xwpf/converter/pdf) do not verify whether the produced pdf retains, for example, super- and subscript texts. as long as the docx can be read and _some_ pdf is produced, everything is ok, which is not always true. in my case,  super- and subscript texts were dropped by the pdf mappers.

**Comment by angelozerr** [comment]
Yes I know @goekay our tests are not really tests because I have not found a solution to do JUNit for that. But the idea is to create a little docx whith a case and after developer needs to check "with her eyes" if he has not broken other cases. I know it's a fragile solution, but I have not found a better solution.

File: thirdparties-extension/fr.opensagres.poi.xwpf.converter.pdf.itext5/src/test/java/org/apache/poi/xwpf/converter/core/AbstractXWPFPOIConverterTest.java [hunk 1]
```diff
@@ -522,6 +522,13 @@ public void TestTitle()
         doGenerate( "TestTitle.docx" );
     }
 
+    @Test
+    public void TestSuperscriptSubscript()
+            throws IOException
+    {
+        doGenerate( "TestSuperscriptSubscript.docx" );
```
Comment (angelozerr) [inline thirdparties-extension/fr.opensagres.poi.xwpf.converter.pdf.itext5/src/test/java/org/apache/poi/xwpf/converter/core/AbstractXWPFPOIConverterTest.java]: Please add TestSuperscriptSubscript.docx in resources folder.

**Comment by angelozerr** [comment]
Thanks @goekay !

**Pull request merged**